src/aiographql/client/client.py

```diff
@@ -271,10 +271,12 @@ async def query(
             )
 
         async with aiohttp_client_session() as session:
-            return await self._http_request(
+            res = await self._http_request(
```

Comment:
Not sure if this is actually required here. I beleive the result will be the same here as the context will be closed when the result is returned. Am I missing something here?

Reply:
At the time when I opened this, I could produce a traceback like this:

```
DEBUG:asyncio:Using selector: EpollSelector
INFO:root:Fetching id for 0xaaaebe6fe48e54f431b0c390cfaf0b017d09d42d from Uniswap V1
DEBUG:asyncio:Get address info api.thegraph.com:443, type=<SocketKind.SOCK_STREAM: 1>, flags=<AddressInfo.AI_ADDRCONFIG: 32>
DEBUG:asyncio:Getting address info api.thegraph.com:443, type=<SocketKind.SOCK_STREAM: 1>, flags=<AddressInfo.AI_ADDRCONFIG: 32> took 22.479ms: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('104.18.11.19', 443)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('104.18.10.19', 443)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('2606:4700::6812:a13', 443, 0, 0)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('2606:4700::6812:b13', 443, 0, 0))]
DEBUG:asyncio:poll 1989.165 ms took 22.023 ms: 1 events
DEBUG:asyncio:connect <socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('0.0.0.0', 0)> to ('104.18.11.19', 443)
DEBUG:asyncio:poll 1965.756 ms took 11.137 ms: 1 events
DEBUG:asyncio:<asyncio.sslproto.SSLProtocol object at 0x7f0b4c888400> starts SSL handshake
DEBUG:asyncio:poll 1952.945 ms took 13.246 ms: 1 events
DEBUG:asyncio:<asyncio.sslproto.SSLProtocol object at 0x7f0b4c888400>: SSL handshake took 15.3 ms
DEBUG:asyncio:<socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.0.0.2', 44070), raddr=('104.18.11.19', 443)> connected to 104.18.11.19:443: (<asyncio.sslproto._SSLProtocolTransport object at 0x7f0b4c888f98>, <aiohttp.client_proto.ResponseHandler object at 0x7f0b4ca14ee8>)
DEBUG:asyncio:poll 1936.199 ms took 202.641 ms: 1 events
DEBUG:asyncio:poll 1731.494 ms took 115.540 ms: 1 events
ERROR:asyncio:Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7f0b4c9da8d0>
source_traceback: Object created at (most recent call last):
  File "test_thegraph.py", line 25, in <module>
    loop.run_until_complete(main())
  File "/usr/lib/python3.7/asyncio/base_events.py", line 571, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.7/asyncio/base_events.py", line 539, in run_forever
    self._run_once()
  File "/usr/lib/python3.7/asyncio/base_events.py", line 1767, in _run_once
    handle._run()
  File "/usr/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "test_thegraph.py", line 12, in main
    data = await thegraph.fetch_uniswap_v1_id("0xaaaebe6fe48e54f431b0c390cfaf0b017d09d42d")
  File "/home/elias/finminity_research_bot/thegraph.py", line 58, in fetch_uniswap_v1_id
    response: GraphQLResponse = await graphql_client.query(request=request)
  File "/home/elias/.local/lib/python3.7/site-packages/aiographql/client/client.py", line 255, in query
    await self.validate(request=request)
  File "/home/elias/.local/lib/python3.7/site-packages/aiographql/client/client.py", line 146, in validate
    schema = schema or await self.get_schema(headers=headers or request.headers)
  File "/home/elias/.local/lib/python3.7/site-packages/aiographql/client/client.py", line 119, in get_schema
    self._schema = await self.introspect(headers=headers)
  File "/home/elias/.local/lib/python3.7/site-packages/aiographql/client/client.py", line 98, in introspect
    introspection = await self.query(request)
  File "/home/elias/.local/lib/python3.7/site-packages/aiographql/client/client.py", line 273, in query
    async with aiohttp_client_session() as session:
  File "/usr/lib/python3.7/contextlib.py", line 170, in __aenter__
    return await self.gen.__anext__()
  File "/home/elias/.local/lib/python3.7/site-packages/aiographql/client/helpers.py", line 21, in aiohttp_client_session
    yield aiohttp.ClientSession(connector=connector)
DEBUG:asyncio:<_SelectorSocketTransport fd=6 read=polling write=<idle, bufsize=0>> received EOF
DEBUG:asyncio:<asyncio.sslproto.SSLProtocol object at 0x7f0b4c888400> received EOF
DEBUG:asyncio:Get address info api.thegraph.com:443, type=<SocketKind.SOCK_STREAM: 1>, flags=<AddressInfo.AI_ADDRCONFIG: 32>
DEBUG:asyncio:Getting address info api.thegraph.com:443, type=<SocketKind.SOCK_STREAM: 1>, flags=<AddressInfo.AI_ADDRCONFIG: 32> took 15.808ms: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('104.18.10.19', 443)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('104.18.11.19', 443)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('2606:4700::6812:a13', 443, 0, 0)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('2606:4700::6812:b13', 443, 0, 0))]
DEBUG:asyncio:poll 1576.591 ms took 15.885 ms: 1 events
DEBUG:asyncio:connect <socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('0.0.0.0', 0)> to ('104.18.10.19', 443)
DEBUG:asyncio:poll 1559.713 ms took 10.881 ms: 1 events
DEBUG:asyncio:<asyncio.sslproto.SSLProtocol object at 0x7f0b47dbe1d0> starts SSL handshake
DEBUG:asyncio:poll 1546.501 ms took 12.947 ms: 1 events
DEBUG:asyncio:<asyncio.sslproto.SSLProtocol object at 0x7f0b47dbe1d0>: SSL handshake took 14.5 ms
DEBUG:asyncio:<socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.0.0.2', 58748), raddr=('104.18.10.19', 443)> connected to 104.18.10.19:443: (<asyncio.sslproto._SSLProtocolTransport object at 0x7f0b47dbe2b0>, <aiohttp.client_proto.ResponseHandler object at 0x7f0b4c9ce3f0>)
DEBUG:asyncio:poll 1531.789 ms took 184.588 ms: 1 events
ERROR:asyncio:Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7f0b47cf7630>
source_traceback: Object created at (most recent call last):
  File "test_thegraph.py", line 25, in <module>
    loop.run_until_complete(main())
  File "/usr/lib/python3.7/asyncio/base_events.py", line 571, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.7/asyncio/base_events.py", line 539, in run_forever
    self._run_once()
  File "/usr/lib/python3.7/asyncio/base_events.py", line 1767, in _run_once
    handle._run()
  File "/usr/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "test_thegraph.py", line 12, in main
    data = await thegraph.fetch_uniswap_v1_id("0xaaaebe6fe48e54f431b0c390cfaf0b017d09d42d")
  File "/home/elias/finminity_research_bot/thegraph.py", line 58, in fetch_uniswap_v1_id
    response: GraphQLResponse = await graphql_client.query(request=request)
  File "/home/elias/.local/lib/python3.7/site-packages/aiographql/client/client.py", line 273, in query
    async with aiohttp_client_session() as session:
  File "/usr/lib/python3.7/contextlib.py", line 170, in __aenter__
    return await self.gen.__anext__()
  File "/home/elias/.local/lib/python3.7/site-packages/aiographql/client/helpers.py", line 21, in aiohttp_client_session
    yield aiohttp.ClientSession(connector=connector)
0x91debb54de02872a259c17699d9b794bc949fed2
DEBUG:asyncio:poll 1345.553 ms took 9.774 ms: 1 events
DEBUG:asyncio:<_SelectorSocketTransport fd=6 read=polling write=<idle, bufsize=0>> received EOF
DEBUG:asyncio:<asyncio.sslproto.SSLProtocol object at 0x7f0b47dbe1d0> received EOF
INFO:asyncio:poll 1335.121 ms took 1337.404 ms: timeout
DEBUG:asyncio:Close <_UnixSelectorEventLoop running=False closed=False debug=True>
```

My patch solved it. But now, I can't reproduce it anymore. Maybe it was another bug in `aiohttp` or somewhere else that has been solved by now. I honestly don't know. Closing this for now, as the bug does not seem to be there anymore.

Reply:
Thank you for the PR though. Apologies for the delay getting to it.